optimum/exporters/tflite/base.py

```diff
@@ -0,0 +1,288 @@
+# coding=utf-8
+# Copyright 2023 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""TensorFlow Lite configuration base classes."""
+
+from abc import ABC, abstractmethod
+from ctypes import ArgumentError
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Tuple, Union
+
+from transformers.utils import is_tf_available
+
+
+if is_tf_available():
+    import tensorflow as tf
+
+from ..base import ExportConfig
+
+
+if TYPE_CHECKING:
+    from transformers import PretrainedConfig, TFPreTrainedModel
+
+    from ...utils import DummyInputGenerator
+
+    if is_tf_available():
+        from tensorflow import TensorSpec
+
+
+class MissingMandatoryAxisDimension(ValueError):
+    pass
```

Comment:
should this be in `error_utils.py`?

Reply:
Not for now. maybe later. but this is quite specific to the `TFLiteConfig`.